File: app/src/main/java/com/kunminx/puremusic/data/repository/DataRepository.java


package com.kunminx.puremusic.data.repository;

import androidx.lifecycle.MutableLiveData;

import com.kunminx.puremusic.R;
import com.kunminx.puremusic.data.bean.GridItem;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

/**
 * Create by KunMinX at 19/10/29
 */
public class DataRepository implements ILocalRequest, IRemoteRequest {

    private static final DataRepository S_REQUEST_MANAGER = new DataRepository();
    private MutableLiveData<String> responseCodeLiveData;

    private DataRepository() {
    }

    public static DataRepository getInstance() {
        return S_REQUEST_MANAGER;
    }

    public MutableLiveData<String> getResponseCodeLiveData() {
        if (responseCodeLiveData == null) {
            responseCodeLiveData = new MutableLiveData<>();
        }
        return responseCodeLiveData;
    }

    @Override
    public void getJavaItems(MutableLiveData<List<GridItem>> liveData) {

        List<GridItem> list = new ArrayList<>();
        list.add(new GridItem(getUUID(), "Lifecycles", R.drawable.bg_album_default, "com.kunminx.jetpack_j2k", "com.kunminx.jetpack_java.sample_01_lifecycles.ui.LifecycleEditorActivity"));
        list.add(new GridItem(getUUID(), "LiveData", R.drawable.bg_album_default, "com.kunminx.jetpack_j2k", "com.kunminx.jetpack_java.sample_02_livedata.ui.LiveDataEditorActivity"));
        list.add(new GridItem(getUUID(), "ViewModel", R.drawable.bg_album_default, "com.kunminx.jetpack_j2k", "com.kunminx.jetpack_java.sample_03_viewmodel.ui.ViewModelListActivity"));
        list.add(new GridItem(getUUID(), "DataBinding", R.drawable.bg_album_default, "com.kunminx.jetpack_j2k", "com.kunminx.jetpack_java.sample_04_databinding.ui.DataBindingListActivity"));
        list.add(new GridItem(getUUID(), "Navigation", R.drawable.bg_album_default, "com.kunminx.jetpack_j2k", "com.kunminx.jetpack_java.sample_05_navigation.ui.NavigationMainActivity"));
        list.add(new GridItem(getUUID(), "One More Thing", R.drawable.bg_album_default, "com.kunminx.jetpack_j2k", "com.kunminx.jetpack_java.sample_one_more_thing.ui.OneMoreThingActivity"));

        liveData.setValue(list);
    }

    @Override
    public void getKotlinItems(MutableLiveData<List<GridItem>> liveData) {

        List<GridItem> list = new ArrayList<>();
        list.add(new GridItem(getUUID(), "Lifecycles", R.drawable.bg_album_default, "com.kunminx.jetpack_j2k", "com.flywith24.jetpack_kotlin.sample_01_lifecycles.ui.LifecycleEditorActivity"));
        list.add(new GridItem(getUUID(), "LiveData", R.drawable.bg_album_default, "com.kunminx.jetpack_j2k", "com.flywith24.jetpack_kotlin.sample_02_livedata.ui.LiveDataEditorActivity"));
        list.add(new GridItem(getUUID(), "ViewModel", R.drawable.bg_album_default, "com.kunminx.jetpack_j2k", "com.flywith24.jetpack_kotlin.sample_03_viewmodel.ui.ViewModelListActivity"));
        list.add(new GridItem(getUUID(), "DataBinding", R.drawable.bg_album_default, "com.kunminx.jetpack_j2k", "com.flywith24.jetpack_kotlin.sample_04_databinding.ui.DataBindingListActivity"));
        list.add(new GridItem(getUUID(), "Navigation", R.drawable.bg_album_default, "com.kunminx.jetpack_j2k", "com.flywith24.jetpack_kotlin.sample_05_navigation.ui.NavigationMainActivity"));
        list.add(new GridItem(getUUID(), "One More Thing", R.drawable.bg_album_default, "com.kunminx.jetpack_j2k", "com.flywith24.jetpack_kotlin.sample_one_more_thing.ui.OneMoreThingActivity"));

        liveData.setValue(list);
    }

    private String getUUID() {
        return UUID.randomUUID().toString().replaceAll("-", "");
    }
}

File: jetpack_kotlin/src/main/java/com/example/jetpack_kotlin/sample_03_viewmodel/data/DataRepository.kt
package com.example.jetpack_kotlin.sample_03_viewmodel.data

import androidx.lifecycle.MutableLiveData
import com.example.jetpack_kotlin.common_data.APIs
import com.example.jetpack_kotlin.common_data.bean.Moment
import java.util.*

/**
 * @author Flywith24
 * @date   2020/5/31
 * time   22:03
 * description
 */
internal class DataRepository private constructor() {

    fun requestList(liveData: MutableLiveData<List<Moment>>) {
        val list = listOf(
                Moment(getUUID(), "刚刚在B站发表了最新一期的视频讲解，感兴趣的小伙伴可前往查阅",
                        "台北夜市一条街", APIs.PIC_URL, "Flywith24", APIs.PIC_URL),
                Moment(getUUID(), "刚刚在B站发表了最新一期的视频讲解，感兴趣的小伙伴可前往查阅",
                        "台北夜市一条街", APIs.PIC_URL, "Flywith24", APIs.PIC_URL),
                Moment(getUUID(), "刚刚在B站发表了最新一期的视频讲解，感兴趣的小伙伴可前往查阅",
                        "台北夜市一条街", APIs.PIC_URL, "Flywith24", APIs.PIC_URL),
                Moment(getUUID(), "刚刚在B站发表了最新一期的视频讲解，感兴趣的小伙伴可前往查阅",
                        "台北夜市一条街", APIs.PIC_URL, "Flywith24", APIs.PIC_URL),
                Moment(getUUID(), "刚刚在B站发表了最新一期的视频讲解，感兴趣的小伙伴可前往查阅",
                        "台北夜市一条街", APIs.PIC_URL, "Flywith24", APIs.PIC_URL),
                Moment(getUUID(), "刚刚在B站发表了最新一期的视频讲解，感兴趣的小伙伴可前往查阅",
                        "台北夜市一条街", APIs.PIC_URL, "Flywith24", APIs.PIC_URL),
                Moment(getUUID(), "刚刚在B站发表了最新一期的视频讲解，感兴趣的小伙伴可前往查阅",
                        "台北夜市一条街", APIs.PIC_URL, "Flywith24", APIs.PIC_URL),
                Moment(getUUID(), "刚刚在B站发表了最新一期的视频讲解，感兴趣的小伙伴可前往查阅",
                        "台北夜市一条街", APIs.PIC_URL, "Flywith24", APIs.PIC_URL)

        )

        liveData.value = list

    }

    private fun getUUID(): String = UUID.randomUUID().toString().replace("-", "")

    companion object {
        private val sRepository = DataRepository()
        fun getInstance(): DataRepository = sRepository
    }
}